### PR TITLE
OpsSight versioning

### PR DESCRIPTION
**Pull request by msenmurugan**
Synopsysctl changes are pending

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/blackduck/components/secret/authcertificate/v1/secret.go [hunk 1]
```diff
@@ -44,23 +45,35 @@ type BdRSecret struct {
...
 		cert, err := stringToCertificate(b.blackDuck.Spec.AuthCustomCA)
 		if err != nil {
-			logrus.Warnf("The Auth Custom CA provided is invalid")
-		} else {
-			logrus.Debugf("Adding The Auth Custom CA with SN: %x", cert.SerialNumber)
-			authCustomCASecret := components.NewSecret(horizonapi.SecretConfig{Namespace: b.blackDuck.Spec.Namespace, Name: apputils.GetResourceName(b.blackDuck.Name, util.BlackDuckName, "auth-custom-ca"), Type: horizonapi.SecretTypeOpaque})
-			authCustomCASecret.AddData(map[string][]byte{"AUTH_CUSTOM_CA": []byte(b.blackDuck.Spec.AuthCustomCA)})
-			authCustomCASecret.AddLabels(apputils.GetVersionLabel("secret", b.blackDuck.Name, b.blackDuck.Spec.Version))
-			secrets = append(secrets, authCustomCASecret)
+			if err != nil {
```
Comment (jdartigalongue) [inline pkg/apps/blackduck/components/secret/authcertificate/v1/secret.go]: No need to check if err != nil twice

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#pullrequestreview-270118708" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/blackduck/components/secret/authcertificate/v1/secret.go (hunk 1 above)
Comment (msenmurugan) [reply pkg/apps/blackduck/components/secret/authcertificate/v1/secret.go]: done

File: pkg/apps/opssight/opssight.go [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+	blackDuckClient *blackduckclientset.Clientset
+	sizeClient      *sizeclientset.Clientset
+	routeClient     *routeclient.RouteV1Client
+	securityClient  *securityclient.SecurityV1Client
+}
+
+// NewOpsSight will return an OpsSight type
+func NewOpsSight(protoformDeployer *protoform.Deployer) *OpsSight {
+	// Initialize the OpsSight client
+	opsSightClient, err := opssightclientset.NewForConfig(protoformDeployer.KubeConfig)
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/opssight/opssight.go]: @yashbhutwala might fix this in separate PR

File: pkg/apps/opssight/components/route/core/v1/coreRouteV1.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/blackducksoftware/synopsys-operator/pkg/util"
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// OpsSightRoute holds the OpsSight Route configuration
+type OpsSightRoute struct {
+	config     *protoform.Config
+	kubeClient *kubernetes.Clientset
+	opsSight   *opssightapi.OpsSight
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/opssight/components/route/core/v1/coreRouteV1.go]: fixed

File: pkg/apps/opssight/components/configmap/opssight/v1/opssightCMV1.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+	}
+
+	cm.AddLabels(map[string]string{"app": "opssight", "component": name, "name": o.opsSight.Name})
+	cm.AddData(map[string]string{fmt.Sprintf("%s.json", o.opsSight.Spec.ConfigMapName): configMapString})
+
+	return cm, nil
+}
+
+// PerceiverConfig stores the Perceiver configuration
+type PerceiverConfig struct {
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/opssight/components/configmap/opssight/v1/opssightCMV1.go]: moved

File: pkg/apps/opssight/components/clusterrolebinding/scanner/v1/scannerCRBV1.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	opsSight, ok := cr.(*opssightapi.OpsSight)
+	if !ok {
+		return nil, fmt.Errorf("unable to cast the interface to OpsSight object")
+	}
+	return &OpsSightClusterRoleBinding{config: config, kubeClient: kubeClient, opsSight: opsSight}, nil
+}
+
+// GetClusterRoleBinding returns the Cluster role binding
+func (o *OpsSightClusterRoleBinding) GetClusterRoleBinding() (*components.ClusterRoleBinding, error) {
+	clusterRole := []string{"synopsys-operator-admin"}
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/opssight/components/clusterrolebinding/scanner/v1/scannerCRBV1.go]: created a new cluster role for scanner but not tested it. will create a separate PR to fix it

File: pkg/apps/opssight/components/clusterrole/imageprocessor/v1/imageprocessorCRV1.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
+/*
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/opssight/components/clusterrole/imageprocessor/v1/imageprocessorCRV1.go]: done

File: pkg/apps/opssight/components/clusterrole/skyfire/v1/skyfire_clusterrole_v1.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// GetClusterRole returns the Cluster Role
+func (o *OpsSightClusterRole) GetClusterRole() (*components.ClusterRole, error) {
+	if !o.opsSight.Spec.EnableSkyfire {
+		return nil, nil
+	}
+
+	clusterRole := components.NewClusterRole(horizonapi.ClusterRoleConfig{
+		Name:       "skyfire",
```
Comment (rrati) [inline pkg/apps/opssight/components/clusterrole/skyfire/v1/skyfire_clusterrole_v1.go]: Shouldn't this be unique too?

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#pullrequestreview-270118708" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#pullrequestreview-270346206" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/opssight/components/clusterrole/skyfire/v1/skyfire_clusterrole_v1.go (hunk 7 above)
Comment (msenmurugan) [reply pkg/apps/opssight/components/clusterrole/skyfire/v1/skyfire_clusterrole_v1.go]: done

File: pkg/apps/opssight/components/clusterrolebinding/skyfire/v1/skyfire_clusterrolebinding_v1.go [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	return &OpsSightClusterRoleBinding{config: config, kubeClient: kubeClient, opsSight: opsSight}, nil
+}
+
+// GetClusterRoleBinding returns the Cluster role binding
+func (o *OpsSightClusterRoleBinding) GetClusterRoleBinding() (*components.ClusterRoleBinding, error) {
+	if !o.opsSight.Spec.EnableSkyfire {
+		return nil, nil
+	}
+	clusterRoleBinding := components.NewClusterRoleBinding(horizonapi.ClusterRoleBindingConfig{
+		Name:       "skyfire",
```
Comment (rrati) [inline pkg/apps/opssight/components/clusterrolebinding/skyfire/v1/skyfire_clusterrolebinding_v1.go]: unique here too?

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#pullrequestreview-270118708" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/724#pullrequestreview-270355357" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**